action.yaml

```diff
@@ -20,5 +20,37 @@ outputs:
     description:
       The version of Terraform that has been installed and is ready for use.
 runs:
-  using: "node20"
-  main: "dist/main/index.js"
+  using: "composite"
+  steps:
+    - name: Read node version
+      id: get-node-version
+      run: |
+        # Check if an .nvmrc file exists and export the node version if it does
+        if [ -f .nvmrc ]; then
+          NODE_VERSION=$(cat .nvmrc)
+          echo "Node version from .nvmrc: $NODE_VERSION"
+          echo "node_version=$NODE_VERSION" >> $GITHUB_OUTPUT
+        fi
+
+        # Check if a .node-version file exists and export the node version if it does
+        if [ -f .node-version ]; then
+          NODE_VERSION=$(cat .node-version)
+          echo "Node version from .node-version: $NODE_VERSION"
+          echo "node_version=$NODE_VERSION" >> $GITHUB_OUTPUT
+        fi
+      shell: bash
+    - uses: actions/setup-node@v4
+      id: setup-node
+      if: steps.get-node-version.outputs.node_version
+      with:
+        node-version: ${{ steps.get-node-version.outputs.node_version }}
```

Comment:
We can optionally provide a GH token if we hit rate limits (https://github.com/actions/setup-node?tab=readme-ov-file#using-setup-node-on-ghes)

We just need to apply this diff:
```
diff --git a/action.yaml b/action.yaml
index 6dcc903..8a78d82 100644
--- a/action.yaml
+++ b/action.yaml
@@ -3,6 +3,11 @@ description:
   GitHub Action that installs and provisions supported tools for workflow steps
   in self-hosted runners. This relies on the agent having supported tooling
   installed.
+inputs:
+  github-token:
+    required: true
+    description: GitHub token to overcome unauthenticated rate limits when installing node, for instance 'secrets.GITHUB_TOKEN'
+    default: ${{ github.token }}
 outputs:
   go:
     description:
@@ -44,6 +49,8 @@ runs:
       if: steps.get-node-version.outputs.node_version
       with:
         node-version: ${{ steps.get-node-version.outputs.node_version }}
+        # https://github.com/actions/setup-node?tab=readme-ov-file#using-setup-node-on-ghes
+        token: ${{ inputs.github-token }}
     - name: Export installed node version
       if: steps.get-node-version.outputs.node_version
       run: |
```